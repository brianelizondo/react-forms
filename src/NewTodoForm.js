import React, { useState } from "react";

function NewTodoForm({ addTask }){
    const INITIAL_STATE = { task: "" };
    const [formData, setFormData] = useState(INITIAL_STATE);

    const handleChange = evt => {
        const { name, value } = evt.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }));
    }
    const handleSubmit = evt => {
        evt.preventDefault();
        addTask(formData);
        setFormData(INITIAL_STATE);
    }

    return (
        <div className="NewTodoForm">
            <form onSubmit={handleSubmit}>
                <label htmlFor="task">Task:</label>
                <input 
                    name="task" 
                    id="task"
                    placeholder="task"
                    value={formData.task} 
                    onChange={handleChange}
                />
                
                <button>Add Task!</button>
            </form>
        </div>
    );
}

export default NewTodoForm;