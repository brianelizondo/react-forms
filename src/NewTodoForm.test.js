import React from "react";
import { render } from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";

// Smoke Tests
it("'NewTodoForm' component renders without crashing", function() {
    render(<NewTodoForm />);
});

// Snapshot Tests
it("'NewTodoForm' matches snapshot", function() {
    const {asFragment} = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
});
