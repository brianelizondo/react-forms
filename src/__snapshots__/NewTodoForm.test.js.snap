// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'NewTodoForm' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="NewTodoForm"
  >
    <form>
      <label
        for="task"
      >
        Task:
      </label>
      <input
        id="task"
        name="task"
        placeholder="task"
        value=""
      />
      <button>
        Add Task!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
